src/kv/store.h

```diff
@@ -759,123 +727,73 @@ namespace kv
         {
-          success = commit_deserialised(changes, v, new_maps, hooks);
-          if (success == DeserialiseSuccess::FAILED)
-          {
-            return success;
-          }
-
-          auto h = get_history();
-          bool result = true;
-          if (sig != nullptr)
-          {
-            auto r = h->verify_and_sign(*sig, term_);
-            if (
-              r != kv::TxHistory::Result::OK &&
-              r != kv::TxHistory::Result::SEND_SIG_RECEIPT_ACK)
-            {
-              result = false;
-            }
-          }
-          else
-          {
-            result = h->verify(term_);
-          }
-
-          if (!result)
-          {
-            LOG_FAIL_FMT("Failed to deserialise");
-            LOG_DEBUG_FMT("Signature in transaction {} failed to verify", v);
-            throw std::logic_error(
-              "Failed to verify signature, view-changes not implemented");
-            return DeserialiseSuccess::FAILED;
-          }
-          h->append(data);
-          success = DeserialiseSuccess::PASS_SIGNATURE;
+          exec = std::make_unique<SignatureBFTExec>(
+            this,
+            get_history(),
+            std::move(data),
+            public_only,
+            v,
+            std::move(changes),
+            std::move(new_maps));
         }
         else if (changes.find(ccf::Tables::BACKUP_SIGNATURES) != changes.end())
         {
-          success = commit_deserialised(changes, v, new_maps, hooks);
-          if (success == DeserialiseSuccess::FAILED)
-          {
-            return success;
-          }
-
-          kv::TxID tx_id;
-
-          auto r = progress_tracker->receive_backup_signatures(
-            tx_id, consensus->node_count(), consensus->is_primary());
-          if (r == kv::TxHistory::Result::SEND_SIG_RECEIPT_ACK)
-          {
-            success = DeserialiseSuccess::PASS_BACKUP_SIGNATURE_SEND_ACK;
-          }
-          else if (r == kv::TxHistory::Result::OK)
-          {
-            success = DeserialiseSuccess::PASS_BACKUP_SIGNATURE;
-          }
-          else
-          {
-            LOG_FAIL_FMT("receive_backup_signatures Failed");
-            LOG_DEBUG_FMT("Signature in transaction {} failed to verify", v);
-            throw std::logic_error(
-              "Failed to verify signature, view-changes not implemented");
-            return DeserialiseSuccess::FAILED;
-          }
-
-          *term_ = tx_id.term;
-          *index_ = tx_id.version;
-
-          auto h = get_history();
-          h->append(data);
+          exec = std::make_unique<BackupSignatureBFTExec>(
+            this,
+            get_history(),
+            get_progress_tracker(),
+            get_consensus(),
+            std::move(data),
+            public_only,
+            v,
+            std::move(changes),
+            std::move(new_maps));
         }
         else if (changes.find(ccf::Tables::NONCES) != changes.end())
```

Comment:
Do we ever check in these branches that there's nothing in the AE but the tables we're looking for? That seems to be done for signatures, and not elsewhere.

Reply:
In store.h:723 we make sure there is only 1 table in changes